mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/ClientLoanIntegrationTest.java

```diff
@@ -4558,5 +4560,140 @@ private Integer createSavingsProduct(final RequestSpecification requestSpec, fin
                 .withMinimumOpenningBalance(minOpenningBalance).build();
         return SavingsProductHelper.createSavingsProduct(savingsProductJSON, requestSpec, responseSpec);
     }
-
+    
+    @Test
+	public void testLoanProductConfiguration() {
+		final String proposedAmount = "5000";
+		JsonObject loanProductConfigurationAsTrue = new JsonObject();
+		loanProductConfigurationAsTrue = this.createLoanProductConfigurationDetailAsTrue(loanProductConfigurationAsTrue);
+		
+		JsonObject loanProductConfigurationAsFalse = new JsonObject();
+		loanProductConfigurationAsFalse = this.createLoanProductConfigurationDetailAsFalse(loanProductConfigurationAsFalse);
+
+		final Integer clientID = ClientHelper.createClient(this.requestSpec,
+				this.responseSpec, "01 January 2012");
+		Integer loanProductID = this.loanTransactionHelper
+				.getLoanProductId(new LoanProductTestBuilder()
+						.withAmortizationTypeAsEqualInstallments()
+						.withRepaymentTypeAsMonth()
+						.withRepaymentAfterEvery("1")
+						.withRepaymentStrategy(
+								LoanProductTestBuilder.MIFOS_STANDARD_STRATEGY)
+						.withInterestTypeAsDecliningBalance()
+						.withInterestCalculationPeriodTypeAsDays()
+						.withInArrearsTolerance("10").withMoratorium("2", "3")
+						.withLoanProductConfiguration(loanProductConfigurationAsTrue).build(null));
+		System.out.println("-----------------------LOAN PRODUCT CREATED WITH ATTRIBUTE CONFIGURATION AS TRUE--------------------------"
+                + loanProductID);
+		Integer loanID = applyForLoanApplicationWithProductConfigurationAsTrue(
+				clientID, loanProductID, proposedAmount);
+		System.out.println("------------------------LOAN CREATED WITH ID------------------------------"
+                + loanID);
+		
+		loanProductID = this.loanTransactionHelper
+				.getLoanProductId(new LoanProductTestBuilder()
+						.withAmortizationTypeAsEqualInstallments()
+						.withRepaymentTypeAsMonth()
+						.withRepaymentAfterEvery("1")
+						.withRepaymentStrategy(
+								LoanProductTestBuilder.MIFOS_STANDARD_STRATEGY)
+						.withInterestTypeAsDecliningBalance()
+						.withInterestCalculationPeriodTypeAsDays()
+						.withInArrearsTolerance("10")
+						.withMoratorium("2", "3")
+						.withLoanProductConfiguration(loanProductConfigurationAsFalse).build(null));
+		System.out.println("-------------------LOAN PRODUCT CREATED WITH ATTRIBUTE CONFIGURATION AS FALSE----------------------"
+                + loanProductID);
+		/* Try to override attribute values in loan account when attribute configurations are set to false at product level */
+		loanID = applyForLoanApplicationWithProductConfigurationAsFalse(clientID, loanProductID, proposedAmount);
+		System.out.println("--------------------------LOAN CREATED WITH ID-------------------------"
+                + loanID);
+		this.validateIfValuesAreNotOverridden(loanID, loanProductID);
+	}
+    
+    private void validateIfValuesAreNotOverridden(Integer loanID, Integer loanProductID){
```

Comment:
too many unnecessary server calls, you can get both loan product and loan attributes in one go and the assert equality unequality etc (can reduce a total of 32 roundtrips to the server when the testcase run)